### ISSUE TITLE
Add support to convert disk provisioning

### ISSUE THREAD

**Issue by vkmellon** [hashicorp/terraform-provider-vsphere#935]
### Terraform Version
0.12.16

### vSphere Provider Version
1.13

### Affected Resource(s)

- `vsphere_virtual_machine`

Hi! I have VMWare template with thin provisioning disk. I want to change disk type on deployment. So I set disks section:
```
  disk {
    label            = "main_drive"
    size             =  60
    unit_number      = 0
    thin_provisioned = false
   eagerly_scrub = true
  }
```
And it is failing with error:
Error: error reconfiguring virtual machine: error processing disk changes post-clone: disk.0: cannot change the value of "thin_provisioned" - (old: true new: false)

### Expected Behavior

Successfully create VM

### Actual Behavior

Error: error reconfiguring virtual machine: error processing disk changes post-clone: disk.0: cannot change the value of "thin_provisioned" - (old: true new: false)

### Steps to Reproduce

1. Create VMWare template with thin_provisioning disk.
2. `terraform apply`

**Comment by bill-rich** [comment]
Currently, the provider does not allow you to change disk types. The reason is that the steps to implement are more of a process than a setting change. Converting from thin to thick is fairly straight forward (as long as there are no snapshots), but converting from thick to thin requires migration between datastores.

I'll put this request on the backlog, and I will look into the API steps required to implement the enhancement.

**Comment by vkmellon** [comment]
@bill-rich 
Thanks for fast response.
Could you please add link to task from backlog?

We are currently choosing tool for deployment VMs on VSphere. Anslible have ability to convert disk types. It blocks me to use Terraform.
Any workaround here?

**Comment by kgorskowski** [comment]
Hey folks,
first of all, thanks for the work that goes into the tool. Highly appreciated, but I wanted to add the problems we face atm.
We want to use terraform and the provider to deploy infrastructure in an automated fashion via gitlab ci. Unfortunately we have to rely on a contractor of our customer to provide access to the vsphere environment and dont manage that ourselves. 
Our initial template is thick provisioned but for some reason, sometimes after a reboot or a vsphere storage vmotion action, the disk of the instance changed to thin, which breaks our terraform state. We then decided to switch to thin provisioning and migrate the already existing instances. But now we run into the problem that we cant "hardcode" thin provisioning in our module because the provisioning type can't be modified at cloning. 
Additionally, after migration, existing vm "sets" can not be scaled up again. 
I get that this is maybe more complex than it seems but at the moment it feels we are running into on wall after another with terraform and (at least our) vsphere environment. 
Not being able to convert provisioning type at cloning could as of now well be a dealbreaker for us unfortunately.

**Comment by idrennanvmware** [comment]
This is a huge blocker for us too.   We have datacenters with storage options that require eager zero but the majority are thin.   Without this ability from a TEMPLATE we now have to maintain 2 separate VM images purely for this scenario

**Comment by greeneg** [comment]
Noted this issue from our side too. Any additional dynamic disks are being created correctly, however, the C drive/root disk of our VM templates are thin provisioned, and need converted at clone to be thick/eager-zeroed. Can you please pull this forward in the backlog?

**Comment by tenthirtyam** [comment]
The context is duplicated in #1028. Closing this issue in favor of issue
#1028.

If you voted 👍 on this issue, please add your vote to #1028. 

TL;DR: The disk provisioning settings are not being applied based on provided arguments given to eagerly_scrubbed and thin_provisioned for a clone.

When cloning a virtual machine, the virtual machine only retains the settings which were originally provisioned.

- thin >> thin
- lazy zero >> lazy zero
- eager zero >> eager zero

**Ryan Johnson**
Staff II Solutions Architect | VMware, Inc.
